### PR TITLE
Update of FFmpeg to n7.1

### PR DESCRIPTION
**Pull request by mdabek-nvidia**
## Category:

Other - dependency update

## Description:

Dependency update to FFmpeg n7.1

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [X] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [X] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

DALI-4077

### CODE REVIEW

**Review by mzient — APPROVED** [review]

File: dali/operators/reader/loader/video/frames_decoder.cc [hunk 1]
```diff
@@ -527,6 +527,7 @@ void FramesDecoder::SeekFrame(int frame_id) {
 
   // Seeking clears av buffers, so reset flush state info
   if (flush_state_) {
+    while (ReadFlushFrame(nullptr, false)) {}
```
Comment (stiepan) [inline dali/operators/reader/loader/video/frames_decoder.cc:530]: Why does it help?

Comment (JanuszL) [reply dali/operators/reader/loader/video/frames_decoder.cc:530]: Apparently `avcodec_flush_buffers` doesn't clean decoder buffers properly anymore, and when we switch to the next file there are frames left in the buffer, and we cannot feed more frames to the decoder without reading them first. So this is an additional mechanism to clean the decoder output buffer as `avcodec_flush_buffers` (or the particular decoder implementation that should do that) doesn't fully work as we expect anymore. When works it should be noop.

Comment (stiepan) [reply dali/operators/reader/loader/video/frames_decoder.cc:530]: Thanks!

Comment (JanuszL) [reply dali/operators/reader/loader/video/frames_decoder.cc:530]: I wish we could narrow it down to a particular change in the codec...

**Review by stiepan — APPROVED** [review]

**Pull request merged**